src/FlatSharp.UnityPolyfills/NativeArray.cs

```diff
@@ -0,0 +1,150 @@
+﻿/*
+ * Copyright 2022 Unity Technologies
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+namespace Unity.Collections
+{
+    public enum NativeArrayOptions
+    {
+        UninitializedMemory = 0,
+        ClearMemory = 1
+    }
+
+    public enum Allocator
+    {
+        Invalid = 0,
+        None = 1,
+        Temp = 2,
+        TempJob = 3,
+        Persistent = 4,
+    }
+
+    public unsafe struct NativeArray<T> : IEnumerable<T>, IEquatable<NativeArray<T>> where T : struct
+    {
+        internal Memory<T> m_Buffer;
+
+        internal void* m_Data;
```

Comment:
There isn't a need to use a pointer here or anything like that. I'd be content with an inner array. I just wanted _some_ way to test this code -- I don't want to mock Unity too closely, and am comfortable assuming that anything matching the correct API surface will work in Unity.